CMakeLists.txt

```
@@ -487,6 +487,51 @@ if (CODECOVERAGE)
   add_link_options(--coverage)
 endif (CODECOVERAGE)
 
+# TAP test support
+option(TAP "Enable TAP test support" ON)
+
+if (TAP)
+  include(FindPerlLibs)
+  find_package(PerlLibs 5.8)
+
+  if (PERLLIBS_FOUND)
+    get_filename_component(PERL_BIN_PATH ${PERL_EXECUTABLE} DIRECTORY)
+
+    find_program(PROVE prove
+      PATHS
+      ${PERL_BIN_PATH}
+      "/usr/bin"
+      NO_DEFAULT_PATH)
```

Not sure there is any difference in reality, but `HINTS` are recommended for "computed" paths (https://cmake.org/cmake/help/latest/command/find_program.html).

> Search the paths specified by the HINTS option. These should be paths computed by system introspection, such as a hint provided by the location of another item already found. Hard-coded guesses should be specified with the PATHS option.

It would be very strange if `perl` was installed but not `prove`, but better safe than sorry.

```suggestion
    find_program(PROVE prove
      HINTS ${PERL_BIN_PATH}
      PATHS "/usr/bin"
      NO_DEFAULT_PATH)
```

---

It might still not be in the standard PATH, so finding the exact location is still worthwhile, especially when there are multiple location. We also want to use the "prove" matching the Perl we found.

---

Yes, hence the "better safe than sorry".